back/tests/SpacesWatcher.test.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { describe, expect, it } from "vitest";
+import { BackToPusherSpaceMessage } from "../src/Messages/generated/messages_pb";
+import { SpacesWatcher } from "../src/Model/SpacesWatcher";
+import { mock } from "vitest-mock-extended";
+import { SpaceSocket } from "../src/SpaceManager";
+import { Writable } from "stream";
+
+describe("SpacesWatcher", () => {
+    it(
+        "should close the socket because no pong was answered to the ping within 20sec",
+        async () => {
+            const eventsWatcher: BackToPusherSpaceMessage[] = [];
+            let isClosed = false;
+            const spaceSocketToPusher = mock<SpaceSocket>({
+                write(chunk: BackToPusherSpaceMessage): boolean {
+                    eventsWatcher.push(chunk);
+                    return true;
+                },
+                end(): ReturnType<Writable["end"]> extends Writable ? SpaceSocket : void {
+                    isClosed = true;
+                    return mock<SpaceSocket>();
+                },
+            });
+
+            const watcher = new SpacesWatcher("uuid-watcher", spaceSocketToPusher);
+            expect(eventsWatcher.some((message) => message.hasPingmessage())).toBe(true);
+
+            await new Promise((resolve) => setTimeout(resolve, 20_000));
+            expect(isClosed).toBe(true);
+        },
+        { timeout: 30_000 }
+    );
```

Comment:
Having a test suite that runs fast is really important. This test, by design, takes 20 seconds.
I don't think we should keep it.
Or if we do, we should put it in a separate "test group" that can be run manually, but that is not run automatically.

Reply:
Or maybe I can set an environment variable to the setInterval of the ping to let it be faster as possible to close the socket if no pong?

Reply:
Yup, that's an idea!

Reply:
Reduced tests to max 300ms, is it better or I need to set it to less? 🥸